pennylane/ops/qubit.py

```diff
@@ -722,6 +724,58 @@ def decomposition(phi, wires):
         return decomp_ops
 
 
+class ControlledPhaseShift(DiagonalOperation):
+    r"""ControlledPhaseShift(phi, wires)
+    A qubit controlled phase shift.
+
+    .. math:: CR_\phi(\phi) = \begin{bmatrix}
+                1 & 0 & 0 & 0 \\
+                0 & 1 & 0 & 0 \\
+                0 & 0 & 1 & 0 \\
+                0 & 0 & 0 & e^{i\phi}
+            \end{bmatrix}.
+
+    .. note:: The first wire provided corresponds to the **control qubit**.
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 1
+    * Gradient recipe: :math:`\frac{d}{d\phi}f(CR_\phi(\phi)) = \frac{1}{2}\left[f(CR_\phi(\phi+\pi/2)) - f(CR_\phi(\phi-\pi/2))\right]`
```

Comment:
I did check this and can confirm that it's unlike the other CRX, CRY, CRZ gates because the 2-parameter shift works:
```python
import pennylane as qml
from pennylane import numpy as np

wires = 2
dev = qml.device("default.qubit", wires=range(wires))

w1 = np.random.random((4, 2, 3))
w2 = np.random.random((4, 2, 3))

@qml.qnode(dev)
def f(x):
    qml.templates.StronglyEntanglingLayers(w1, wires=range(wires))
#     qml.CRX(x, wires=[0, 1])
    qml.ControlledPhaseShift(x, wires=[0, 1])
    qml.templates.StronglyEntanglingLayers(w2, wires=range(wires))
    return qml.expval(qml.PauliZ(0))

x = 0.3
g1 = qml.grad(f)(x)
g2 = (f(x + np.pi / 2) - f(x - np.pi / 2)) / 2

np.allclose(g1, g2)
```